DearPyGui/src/core/Theming/mvTheme.cpp

```diff
@@ -1,17 +1,45 @@
 #include "mvTheme.h"
-#include "mvAppItemTheme.h"
 
 namespace Marvel {
 
 	mvTheme::mvTheme()
 	{
-		for (int i = 1; i < (int)mvAppItemType::ItemTypeCount; i++)
-			m_themeItems[(mvAppItemType)i] = mvAppItemTheme(i);
+		mvEventBus::Subscribe(this, SID("color_change"), mvEVT_CATEGORY_THEMES);
 	}
 
-	mvAppItemTheme* mvTheme::getThemeItem(mvAppItemType type)
+	mvTheme::~mvTheme()
 	{
-		return &m_themeItems.at(type);
+		mvEventBus::UnSubscribe(this);
+	}
+ 
+	bool mvTheme::onEvent(mvEvent& event)
+	{
+		mvEventDispatcher dispatcher(event);
+		dispatcher.dispatch(BIND_EVENT_METH(mvTheme::add_color), SID("color_change"), mvEVT_CATEGORY_THEMES);
+		return event.handled;
+	};
+
+	bool mvTheme::add_color(mvEvent& event)
+	{
+		mvAppItemType type = (mvAppItemType)(GetEInt(event, "ID") / 100);
+		int libID = GetEInt(event, "ID") % 100;
+		mvColor color = GetEColor(event, "COLOR");
+		const std::string& widget = GetEString(event, "WIDGET");
+
+		//this needs to fill out the root or primary window
+		if (widget.empty())
+		{
+			widget == mvApp::GetApp()->getItemRegistry().getWindows()[0];
+		}
+
+		//check widget can take color and apply
+		mvRef<mvAppItem> item = mvApp::GetApp()->getItemRegistry().getItem(widget);
+		if (item->getDescription().container || item->getType() == type)
+		{
+			item->getColors()[type][libID] = color;
+		}
+		else ThrowPythonException("Item does not except this theme constant.");
```

Comment:
I'm not sure yet if this is going to be an issue. If this is ran on the main thread, it will crash (since python calls need to be in the python thread). If we every plan on calling this in the main thread (for example the default theme), then we can't use the "ThrowPythonException" command.